### DISCUSSION TITLE
How can I test MutableSharedFlow<Unit>() with turbine ?

### DISCUSSION THREAD

**Discussion by ismailtosun** [cashapp/turbine#116] [Q&A]
Hi,
when i press button MutableSharedFlow<Unit>() is emiting. So how can i test this with turbine emiting or not ? 

// in ViewModel 
private val _updateAddress = MutableSharedFlow<Unit>()
val updateAddress = _updateAddress.asSharedFlow()

fun updateAddressClick() {
        viewModelScope.launch {
            _updateAddress.emit(Unit)
        }
    }

//Test Class
@Test
fun updateAddressBtn_clickEvent_Emiting_or_Not() = runTest {
        viewModel.updateAddress.test {
            // given
            // when
            viewModel.updateAddressClick()
            // then
            Truth.assertThat(awaitEvent())
        }
    }

**Comment by JakeWharton** [comment]
What you wrote looks fine. There is no real way to test the absence of an emission since your code could emit an item a millisecond later. All you can assert in the negative is that no events have been seen up until the current point in time.
